### PR TITLE
Add script to quantize ONNX model

### PR DESCRIPTION
**Pull request by take-cheeze**
https://github.com/microsoft/onnxruntime/tree/master/onnxruntime/python/tools/quantization

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -17,3 +17,6 @@
 [submodule "third_party/json"]
 	path = third_party/json
 	url = https://github.com/nlohmann/json
+[submodule "third_party/onnxruntime"]
```
Comment (shinh) [inline .gitmodules]: Do you think we will use more components in onnxruntime? I guess we'll unlikely do? If it's the case, I'd prefer not submoduling it. As onnxruntime is fairly big (especially with recursive clone), I'm afraid it'd slow down CI or developers. Also, I'm not sure, but I'm guessing their plan is to make the quantize module available from pip (probably by contributing it to ONNX?).

Comment (take-cheeze) [reply .gitmodules]: I'm concerning that too.
Currently downloading [quantize.py](https://github.com/microsoft/onnxruntime/blob/2698edbc98950c31edbae91af8ca049f61c46eb6/onnxruntime/python/tools/quantization/quantize.py) with curl seems enough for now so will fix this.

**Comment by take-cheeze** [comment]
Example running quantized mnist:
```bash
./utils/quantize.sh --no_per_channel ./data/mnist/model.onnx ./data/mnist/q_model.onnx
./build/tools/run_onnx --compiler_log --test=./data/mnist --onnx=./data/mnist/q_model.onnx
```

```
Quantize config: {'nbits': 8, 'quantization_mode': 0, 'static': False, 'asymmetric_input_types': False, 'input_quantization_params': None, 'output_quantization_params': None, 'per_channel': False}
Saving "./data/mnist/model.onnx" to "./data/mnist/q_model.onnx"
Initializing ChainerX...
Loading model...
Loading data...
Found 3 test cases
Constructing model...
ReduceMin simplified
ReduceMin simplified
ReduceMin simplified
ReduceMin simplified
Simulated memory usage: param=0MB peak=0MB all=0MB
Total flops: 64207
Generate code...
Running for ./data/mnist/test_data_set_0
Verifying the result...
FAIL(value): Plus214_Output_0
Expected: array([[  975.67010498,  -618.72393799,  6574.56835938,   668.02893066,  -917.27093506, -1671.63586426, -1952.75988770,   -61.54987335,  -777.17663574, -1439.53161621]], shape=(1, 10), dtype=float32, device='native:0')
Actual: array([[  970.61535645,  -610.13140869,  6588.47851562,   625.06878662,  -911.78411865, -1684.60876465, -1983.74023438,   -29.74928093,  -788.45288086, -1440.35058594]], shape=(1, 10), dtype=float32, device='native:0')
Mismatch: 10 / 10 (100%)
```

**Comment by take-cheeze** [comment]
Seems MaxPool is failing due to ONNX's new op...

**Comment by take-cheeze** [comment]
ChainerCI, test this please!

File: tools/run_onnx.cc [hunk 2]
```diff
@@ -558,10 +558,7 @@ void RunMain(const std::vector<std::string>& argv) {
     std::string test_path = args.get<std::string>("test");
 
     g_quiet = args.exist("quiet");
-    if (!onnx_path.empty() && !test_path.empty()) {
-        std::cerr << args.usage() << std::endl;
-        QFAIL() << "Specifying both --onnx and --test is invalid!";
-    } else if (onnx_path.empty() && test_path.empty()) {
+    if (onnx_path.empty() && test_path.empty()) {
```
Comment (shinh) [inline tools/run_onnx.cc]: What is the intention of this change?

File: runtime/chainerx_util.cc [hunk 3]
```diff
@@ -229,8 +229,26 @@ chainerx::Array GroupedConv(
...
         const Int64StackVector& strides,
-        const Int64StackVector& pads,
-        int group) {
+        const Int64StackVector& in_pads,
+        int group,
+        const std::string& auto_pad) {
+    Int64StackVector pads = in_pads;
+    if (auto_pad == "SAME_UPPER") {
+        for (size_t i = 0; i < pads.size(); ++i) {
+            int64_t& pad = pads[i];
```
Comment (shinh) [inline runtime/chainerx_util.cc]: Unnecessary variable?

File: tools/run_onnx.cc (hunk 2 above)
Comment (take-cheeze) [reply tools/run_onnx.cc]: I want to use test data with the quantized model separately from the original non-quantized model

Comment (shinh) [reply tools/run_onnx.cc]: Ah, that sounds very useful. I'm not sure why I haven't thought of this, thanks!

**Comment by take-cheeze** [comment]
ChainerCI, test this please!

**Pull request merged**